File: WinForms1/WinFormsExample/Form1.vb
Public Class Form1
    '19 Variant Task 1'
    Private Sub calculateButton_Click(sender As Object, e As EventArgs) Handles calculateButton.Click
        Try
            Dim angleA As Double
            Dim angleB As Double
            Dim sideC As Double

            If Not Double.TryParse(textBoxA.Text, angleA) Then
                MessageBox.Show("Угол A должен быть числом.")
                Return
            End If

            If Not Double.TryParse(textBoxB.Text, angleB) Then
                MessageBox.Show("Угол B должен быть числом.")
                Return
            End If

            If Not Double.TryParse(textBoxC.Text, sideC) Then
                MessageBox.Show("Сторона c должна быть числом.")
                Return
            End If

            If angleA <= 0 Then
                MessageBox.Show("Угол A должен быть положительным числом.")
                Return
            End If

            If angleB <= 0 Then
                MessageBox.Show("Угол B должен быть положительным числом.")
                Return
            End If

            If angleA + angleB >= 180 Then
                MessageBox.Show("Сумма углов A и B должна быть меньше 180°.")
                Return
            End If

            If sideC <= 0 Then
                MessageBox.Show("Сторона c должна быть положительным числом.")
                Return
            End If

            Dim angleC As Double = 180 - angleA - angleB

            Dim angleARad As Double = angleA * Math.PI / 180
            Dim angleBRad As Double = angleB * Math.PI / 180
            Dim angleCRad As Double = angleC * Math.PI / 180

            Dim sideA As Double = (sideC * Math.Sin(angleARad)) / Math.Sin(angleCRad)
            Dim sideB As Double = (sideC * Math.Sin(angleBRad)) / Math.Sin(angleCRad)

            textBoxResultAngleC.Text = angleC.ToString()
            textBoxResultSideA.Text = sideA.ToString()
            textBoxResultSideB.Text = sideB.ToString()

        Catch ex As Exception
            MessageBox.Show("Ошибка: " & ex.Message)
        End Try
    End Sub
End Class

File: Lab1/Lab1_Task1/Lab1_Task1/Form1.cs
using System;
using System.Windows.Forms;

namespace Lab1_Task1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            calculateButton.Click += calculateButton_Click;
        }

        private void Form1_Load(object sender, EventArgs e)
        {
        }

        private void calculateButton_Click(object sender, EventArgs e)
        {
            try
            {
                double angleA;
                double angleB;
                double sideC;

                if (!double.TryParse(textBoxA.Text, out angleA))
                {
                    MessageBox.Show("Angle A must be a number.");
                    return;
                }

                if (!double.TryParse(textBoxB.Text, out angleB))
                {
                    MessageBox.Show("Angle B must be a number.");
                    return;
                }

                if (!double.TryParse(textBoxC.Text, out sideC))
                {
                    MessageBox.Show("Side c must be a number.");
                    return;
                }

                if (angleA <= 0)
                {
                    MessageBox.Show("Angle A must be a positive number.");
                    return;
                }

                if (angleB <= 0)
                {
                    MessageBox.Show("Angle B must be a positive number.");
                    return;
                }

                if (angleA + angleB >= 180)
                {
                    MessageBox.Show("The sum of angles A and B must be less than 180°.");
                    return;
                }

                if (sideC <= 0)
                {
                    MessageBox.Show("Side c must be a positive number.");
                    return;
                }

                double angleC = 180 - angleA - angleB;
                double angleARad = angleA * Math.PI / 180;
                double angleBRad = angleB * Math.PI / 180;
                double angleCRad = angleC * Math.PI / 180;

                double sideA = (sideC * Math.Sin(angleARad)) / Math.Sin(angleCRad);
                double sideB = (sideC * Math.Sin(angleBRad)) / Math.Sin(angleCRad);

                textBoxResultAngleC.Text = angleC.ToString("F2");
                textBoxResultSideA.Text = sideA.ToString("F2");
                textBoxResultSideB.Text = sideB.ToString("F2");
            }
            catch (Exception ex)
            {
                MessageBox.Show("Error: " + ex.Message);
            }
        }
    }
}
